x-pack/plugins/lens/public/xy_visualization/visualization.tsx

```diff
@@ -527,6 +527,83 @@ export const getXyVisualization = ({
     };
   },
 
+  updateLayersConfigurationFromContext({ prevState, layerId, context }) {
+    const { chartType, axisPosition, palette, metrics } = context;
+    const foundLayer = prevState?.layers.find((l) => l.layerId === layerId);
+    if (!foundLayer) {
+      return prevState;
+    }
+    const axisMode = axisPosition as YAxisMode;
+    const yConfig = metrics.map((metric, idx) => {
+      return {
+        color: metric.color,
+        forAccessor: metric.accessor ?? foundLayer.accessors[idx],
+        ...(axisMode && { axisMode }),
+      };
+    });
+    const newLayer = {
+      ...foundLayer,
+      ...(chartType && { seriesType: chartType as SeriesType }),
+      ...(palette && { palette }),
+      yConfig,
+    };
+
+    const newLayers = prevState.layers.map((l) => (l.layerId === layerId ? newLayer : l));
+
+    return {
+      ...prevState,
+      layers: newLayers,
+    };
+  },
+
+  getVisualizationSuggestionFromContext({ suggestions, context }) {
+    const visualizationStateLayers = [];
+    let datasourceStateLayers = {};
+    const fillOpacity = context.configuration.fill ? Number(context.configuration.fill) : undefined;
+    for (let suggestionIdx = 0; suggestionIdx < suggestions.length; suggestionIdx++) {
+      const currentSuggestion = suggestions[suggestionIdx] as XYSuggestion;
```

Comment:
There's a lot of `as` in this function... I wonder if instead we could define `getVisualizationSuggestionFromContext` `suggestions` param as `Array<Suggestion<XYVisualization>>` and adjust the interface to it? One example of this pattern is how we handle index pattern suggestions (search for `DatasourceSuggestion<IndexPatternPrivateState>`)

Reply:
Discussed with Marta offline. This will require a lot of changes in many Lens files so maybe it not worth it atm or at this PR. We can always revisit it later.